testing/conformance/benches/bench_drivers.rs

```diff
@@ -0,0 +1,157 @@
+/// Currently needs some serious speedup, probably with respect to WASM caching and also machine setup/teardown.
+pub fn bench_vector_variant(
+    group: &mut BenchmarkGroup<measurement::WallTime>,
+    name: String,
+    variant: &Variant,
+    vector: &MessageVector,
+    messages_with_lengths: Vec<(Message, usize)>,
+    bs: &MemoryBlockstore,
+    engine: &Engine,
+) {
+    group.bench_function(name, move |b| {
+        b.iter_batched_ref(
+            || {
+                let vector = &(*vector).clone();
+                let bs = bs.clone();
+                // TODO next few lines don't impact the benchmarks, but it might make them run waaaay more slowly... ought to make a base copy of the machine and exec and deepcopy them each time.
+                let machine = TestMachine::new_for_vector(vector, variant, bs, engine.clone());
+                // can assume this works because it passed a test before this ran
+                machine.load_builtin_actors_modules().unwrap();
+                let exec: DefaultExecutor<TestKernel> = DefaultExecutor::new(machine);
+                (messages_with_lengths.clone(), exec)
+            },
+            |(messages, exec)| apply_messages(criterion::black_box(messages), exec),
+            BatchSize::LargeInput,
+        )
+    });
+}
+/// This tells `bench_vector_file` how hard to do checks on whether things succeed before running benchmark
+#[allow(dead_code)]
+#[derive(Clone, Copy, Debug)]
+pub enum CheckStrength {
+    /// making sure everything conforms before benching, for when you're benching the real vector as it came from specs-actors
+    FullTest,
+    /// use in cases where we're swapping out the messages to apply and just using the setup (overhead tests, for example)
+    OnlyCheckSuccess,
+    /// use if for some reason you want to bench something that errors (or go really fast and dangerous!)
+    NoChecks,
+}
+
+/// default is FullTest
+impl Default for CheckStrength {
+    fn default() -> Self {
+        CheckStrength::FullTest
+    }
+}
+
+#[derive(Default)]
+/// configuration for all the various tweaks in ways you might want to bench a given vector file
+pub struct BenchVectorFileConfig {
+    /// should only the first variant be run, or all of them?
+    pub only_first_variant: bool,
+    /// should we check whether the vector executes correctly or just without error before benching, or even run no checks at all?
+    pub check_strength: CheckStrength,
+    /// optionally, should we replace the messages to apply here? useful when you just want to pull out the initial FVM setup and run something else.
+    pub replacement_apply_messages: Option<Vec<ApplyMessage>>,
+    /// name for the benchmark as stored on disk.
+    pub bench_name: String,
+}
```

Comment:
Free-flow comments are great inside _code_ as "this is what I was thinking when I wrote this" breadcrumbs, but not as documentation.